### DISCUSSION TITLE
how to pass an expression from PyCall

### DISCUSSION THREAD

**Discussion by randomx207** [MilesCranmer/PySR#533] [Q&A]
Hi @MilesCranmer, 

I would also like to ask: what should I do if I want to generate an initial population (i.e., a sympy expression list, or preorder traversal token list, whatever) in Python, and pass it to julia during the pysr population initialization phase? I have looked at https://github.com/MilesCranmer/PySR/discussions/491 and have clone the source code, but I want to know:
1. Is it possible to modify the function `function Population` to pass expressions from PyCall?
2. If so, how do I build a pysr tree from python's expression objects? I don't know julia well. Can you help me?

for example, what I want to do is like this

```
    return Population{T,L}(
        [
            PopMember(
                dataset,
                call_python_and_get_a_tree(),       # here
                options;
                parent=-1,
                deterministic=options.deterministic,
            ) for i in 1:population_size
        ],
```

**Comment by MilesCranmer** [comment]
This is tricky, I guess the main thing you can do is modify the `PopMember` constructor. But other than that there isn’t currently an easy way to specify the way new equations are generated in a fresh population. Maybe also look at https://github.com/MilesCranmer/SymbolicRegression.jl/issues/193 for what I want to do.

For constructing trees, take a look at DynamicExpressions.jl which is how the node type is defined.
